spec/index.bs

```diff
@@ -2396,6 +2397,29 @@ namespace GPUTextureUsage {
 };
 </script>
 
+<dl dfn-type=dict-member dfn-for=GPUTextureDescriptor>
+    : <dfn>viewFormats</dfn>
+    ::
+        Specifies what view {{GPUTextureViewDescriptor/format}} values will be allowed when calling
+        {{GPUTexture/createView()}} on this texture.
+
+        <!-- This could be extended in the future with either more enum options
```

Comment:
Why not go for it from the start? Currently, the PR defines this new layer of abstraction by introducing the ways of "modifying" formats, like adding/removing sRGB. So users have to think which exact formats they can use, anyway. They might as well list these formats. If they are wrong, they'll get an error earlier.

Reply:
Because we don't have consensus on whether to do general reinterpretation with a flag or a list, but we had tentative consensus on adding something for srgb reinterpretation specifically.

Reply:
(#168)

Reply:
Ok, I see. On one hand, we have to have spec text similar to this describing which classes of reinterpretations are going to be allowed, regardless of how the API is actually shaped at view creation. So just making this description of classes to be the API makes sense.

But on the other hand, it makes less sense. As a user of WebGPU, I know a texture format, and I know which one I'll need to reinterpret with. So instead of just telling it, I have to go through this list of classes and see which one includes my format conversion. Then the implementation has to potentially do the opposite on Vulkan.

Speaking of which, would the implementation pessimistically specify all of the possible formats (allowed by selected classes) to `VK_KHR_image_format_list`? Do we have an idea if this will make it less efficient than telling Vulkan the exact format we want to use?

Reply:
With `VK_KHR_image_format_list` I'm not sure. In the srgb case it's better than a general 'reinterpret' flag since we only have to list the srgb formats when using that extension on the backend. But if we had a general 'reinterpret' flag that was based on the intersection of Vulkan/D3D12/Metal compatibility classes then I don't know.

Reply:
Right, the "srgb" is a simpler case because it only has 2 formats in a class. So an implementation on `VK_KHR_image_format_list` would know exactly what to specify, and it would be no worse than the user telling us the list of formats.

However, this `GPUTextureReinterpretationOption` enum is meant to grow into other things, which are going to be more wide. So it's not clear to me how it's going to scale, given the concerns here. Perhaps, it would be productive to come up with at least one more value for it, so that we can see it more generally?

Reply:
The other option I have in mind would be a predefined list depending on the texture format, according to the investigation https://github.com/gpuweb/gpuweb/issues/744#issuecomment-644481142. So essentially everything that's green on the D3D12 tab of that spreadsheet as it's the most restrictive. It could potentially be called something like "same-layout" (D3D12 requires the components sit in the same bits of the texel, but allows reinterpreting between int/unorm/float/etc).

However if we ended up agreeing on a format list then perhaps we wouldn't bother with that.

Reply:
So there can be:
  - srgb (current)
  - same layout (the green boxes in D3D12 table), which also includes srgb
  - 32bit-uint (for the yellow box in D3D12 table)

Reply:
The yellow box is a bit complex because it only works for UAVs. I'm not sure how we would want to expose that.